hadoop-crypto-api/src/main/java/com/palantir/hadoop/KeyMaterials.java

```diff
@@ -56,9 +56,9 @@ public static SecretKey generateKey(String keyAlgorithm, int keySize) {
     }
 
     public static byte[] generateIv(int ivSize) {
+        OpensslSecureRandom random = new OpensslSecureRandom();
         byte[] iv = new byte[ivSize];
-        SecureRandom rng = new SecureRandom();
-        rng.nextBytes(iv);
+        random.nextBytes(iv);
```

Comment:
There is no particular performance hunt going on - Steve is poking at this because he just fixed the YARN/Hadoop dists to send native libs everywhere and I wanted to make sure we could use them in our S3 encryption as well.